src/main.ts

```diff
@@ -0,0 +1,62 @@
+import { abciQuery } from './abci-query';
+
+const fetchRPCAddr = async (netconfigURL: string) => {
+  const response = await fetch(netconfigURL, {
+    headers: { accept: 'application/json' },
+  });
+  const { rpcAddrs } = await response.json();
+
+  return rpcAddrs[Math.floor(Math.random() * rpcAddrs.length)];
+};
+
+const networkConfigUrl = (() => {
+  const agoricNetName =
+    new URLSearchParams(window.location.search).get('network') ?? 'main';
+
+  if (agoricNetName === 'local') {
+    return 'https://wallet.agoric.app/wallet/network-config';
+  } else {
+    return `https://${agoricNetName}.agoric.net/network-config`;
+  }
+})();
+
+const setMessage = (message: string) => {
+  document?.getElementById('msg')?.replaceChildren(message);
+  console.info(message);
+};
+
+const tryRedirect = async () => {
+  const rpc = await fetchRPCAddr(networkConfigUrl);
+  let endorsedUI;
+
+  try {
+    const data = await abciQuery(
```

Comment:
@turadg I got something working but I'm a little puzzled by the serialization of the response

Reply:
But yay!
```
dist/index.html                 0.80 kB
dist/assets/index-e9cc255b.css  5.07 kB │ gzip: 1.78 kB
dist/assets/index-da439c6c.js   1.90 kB │ gzip: 1.01 kB
```

Reply:
Double yay! I just noticed the merged PR was only into this one. Why have a separate PR?

Anyway, I do think a `vstorageQuery` would be worth abstracting, to keep `:published.vaultFactory.governance` as the const somewhere for the source of this data.

Reply:
I wanted to push it somewhere before changing this PR because I switched computers, I could've cherry-picked instead of merged I guess.